presentation/backend/src/presentation-backend/RulesetEmbedder.ts

```diff
@@ -17,16 +18,42 @@ import {
 import { Ruleset } from "@bentley/presentation-common";
 import { PresentationRules } from "./domain/PresentationRulesDomain";
 import * as RulesetElements from "./domain/RulesetElements";
+import { isEnum, normalizeVersion } from "./Utils";
 
 /**
- * Available strategies for handling duplicate rulesets.
+ * Strategies for handling duplicate rulesets.
  * @beta
+ * @deprecated Use [[RulesetInsertOptions]]
  */
 export enum DuplicateRulesetHandlingStrategy {
+  /** Do not insert the ruleset if another ruleset with the same ID already exists. */
   Skip,
+
+  /** Replace already existing ruleset if one exists with the same ID and version. */
   Replace,
 }
 
+/**
+ * Options for [[RulesetEmbedder.insertRuleset]] operation.
+ * @beta
+ */
+export interface RulesetInsertOptions {
+  /**
+   * When should insertion be skipped:
+   * - `same-id` - if iModel already contains a ruleset with the same id and **any** version
+   * - `same-id-and-version` - if iModel already contains a ruleset with same if and version
+   *
+   * Defaults to `same-id-and-version`.
+   */
+  skip?: "never" | "same-id" | "same-id-and-version";
+
+  /**
+   * Which existing rulesets should be replaced when we insert a new one.
+   * Defaults to `same-id-and-version`.
+   */
+  replace?: "same-id" | "same-id-older" | "same-id-and-version";
+}
+

```

Comment:
I haven't looked too deeply into this, but at first glance it feels like `skip` and `replace` options overlap. What should happen when we specify both as `same-id`?

Reply:
Yes, they overlap (e.g. in your given case, if a ruleset with same id is found, then we just skip insertion and don't even use at `replace` value). 

However, there are cases when both are needed, e.g. when we want to skip inserting if there's a ruleset with same id and version, but if it's not - we want to specify what to replace: all rulesets with same id / only older rulesets. And I didn't want to add long values like `skip-on-same-id-and-version-and-replace-all`, `skip-on-same-id-and-version-and-replace-older`, etc..

Reply:
If only rulesets with the same id are always going to get replaced, how about documenting this fact and renaming options to `"all" | "all-smaller" | "exact"`?

Reply:
Made suggested change. I chose to rename `replace` to `replaceVersions` to make it clear that all options are based on ruleset version and ID is already accounted for.

Reply:
Thanks